data_integration/notification/slack.py

```diff
@@ -1,91 +1,77 @@
-
-        elif isinstance(event, pipeline_events.NodeFinished):
-            key = tuple(event.node_path)
-            if not event.succeeded and event.is_pipeline is False:
-
-                message = {'text': '\n:baby_chick: Ooops, a hiccup in '
-                                   + '_ <' + config.base_url() + '/' + '/'.join(event.node_path)
-                                   + ' | ' + '/'.join(event.node_path) + ' > _',
-                           'attachments': []}
-
-                if (self.node_output[key][False]):
-                    message['attachments'].append({'text': self.format_output(self.node_output[key][False]),
-                                                   'mrkdwn_in': ['text']})
-
-                if (self.node_output[key][True]):
-                    message['attachments'].append({'text': self.format_output(self.node_output[key][True]),
-                                                   'color': '#eb4d5c', 'mrkdwn_in': ['text']})
-
-                response = requests.post('https://hooks.slack.com/services/' + config.slack_token(), json=message)
-
-                if response.status_code != 200:
-                    raise ValueError(
-                        'Request to slack returned an error %s. The response is:\n%s' % (
-                            response.status_code, response.text)
-                    )
-        elif isinstance(event, pipeline_events.RunStarted):
-            # default handler only handles interactively started runs
-            if event.interactively_started:
-                message = f':hatching_chick: *{event.user}* manually triggered run of '
-                message += ('pipeline <' + config.base_url() + '/' + '/'.join(event.node_path) + '|'
-                            + '/'.join(event.node_path) + ' >' if not event.is_root_pipeline else 'root pipeline')
-
-                if event.node_ids:
-                    message += ', nodes ' + ', '.join([f'`{id_}`' for id_ in event.node_ids])
-
-                requests.post('https://hooks.slack.com/services/' + config.slack_token(), json={'text': message})
-        elif isinstance(event, pipeline_events.RunFinished):
-            # default handler only handles interactively started runs
-            if event.interactively_started:
-                if event.succeeded:
-                    msg = ':hatched_chick: succeeded'
-                else:
-                    msg = ':baby_chick: failed'
-                requests.post('https://hooks.slack.com/services/' + config.slack_token(),
-                              json={'text': msg})
+from data_integration.notification.notifier import ChatNotifier
+
+
+class Slack(ChatNotifier):
+
+    def __init__(self):
+        super().__init__()
+
+    def send_run_started_interactively_message(self, event: pipeline_events.RunStarted):
+        text = (':hatching_chick: *' + (os.environ.get('SUDO_USER') or os.environ.get('USER') or os.getlogin())
```

Comment:
The original code was:

```py
                message = f':hatching_chick: *{event.user}* manually triggered run of '
                message += ('pipeline <' + config.base_url() + '/' + '/'.join(event.node_path) + '|'
                            + '/'.join(event.node_path) + ' >' if not event.is_root_pipeline else 'root pipeline')

                 if event.node_ids:
                    message += ', nodes ' + ', '.join([f'`{id_}`' for id_ in event.node_ids])
```

Reply:
@jankatins , you are right.
We have information of user in event now.